orchestra/workflow/load.py

```diff
@@ -0,0 +1,161 @@
+    for certification_data in workflow_data['certifications']:
+        create_or_update(
+            Certification,
+            {
+                'slug': certification_data['slug'],
+                'workflow': workflow
+            },
+            {
+                'name': certification_data['name'],
+                'description': certification_data['description'],
+            }
+        )
+
+    # Create the certification dependencies once all certs are in the db
+    for certification_data in workflow_data['certifications']:
+        cert = Certification.objects.get(
+            slug=certification_data['slug'],
+            workflow=workflow
+        )
+        required_cert_slugs = certification_data['required_certifications']
+        required_certs = Certification.objects.filter(
+            workflow=workflow,
+            slug__in=required_cert_slugs
+        )
+        if required_certs.count() != len(required_cert_slugs):
+            raise WorkflowError(
+                'Certification {} requires non-existent certification.'
+                .format(certification_data['slug']))
+        cert.required_certifications = list(required_certs)
+
+    # Load the desired versions
+    seen_versions = set()
+    requested_versions = set(versions)
+    for version_data in data['versions']:
+        version_slug = version_data['slug']
+        seen_versions.add(version_slug)
+        if versions is None or version_slug in requested_versions:
+            load_workflow_version(version_data, workflow, force=force)
+    missing_versions = requested_versions - seen_versions
+    if versions is not None and missing_versions:
+        raise WorkflowError('Invalid versions requested: {}'
+                            .format(missing_versions))
+
+
+def load_workflow_version(version_data, workflow, force=False):
+    # Create the version object
+    version, version_created = create_or_update(
+        WorkflowVersion,
+        {
+            'slug': version_data['slug'],
+            'workflow': workflow
+        },
+        {
+            'name': version_data['name'],
+            'description': version_data['description']
+        }
+    )
+
+    if not version_created:
+        if not force:
+            raise WorkflowError('Version {} already exists'
+                                .format(version_data['slug']))
+
+        # Check that the versions are safe to merge
+        new_step_slugs = set(s['slug'] for s in version_data['steps'])
+        old_step_slugs = set(
+            WorkflowStep.objects
+            .filter(workflow_version=version)
+            .values_list('slug', flat=True)
+        )
+        if new_step_slugs != old_step_slugs:
+            raise WorkflowError('Even with --force, cannot change the steps '
+                                'of a workflow. Drop and recreate the '
+                                'database to reset, or create a new version '
+                                'for your workflow.')
```

Comment:
negative example: if there are no projects for that workflow version, should we give them a free pass?  i'd opt for no, since a developer can always just delete those objects themselves